sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -1312,6 +1312,101 @@ class SumBinaryFixer(toType: DataType, isAnsi: Boolean)
   }
 }
 
+/**
+ * Fixes up a sum operation for unbounded preceding to unbounded following
+ * @param errorOnOverflow if we need to throw an exception when an overflow happens or not.
+ */
+class SumUnboundedToUnboundedFixer(resultType: DataType, failOnError: Boolean)
+    extends BatchedUnboundedToUnboundedWindowFixer {
+
+  private var previousValue: Option[Scalar] = None
+
+  override def updateState(scalar: Scalar): Unit = {
+    if (scalar.isValid) {
+      previousValue match {
+        case None =>
+          previousValue = Some(scalar.incRefCount())
+        case Some(prev) =>
+          withResource(prev) { _ =>
+            previousValue = None
+            prev.getType.getTypeId match {
+              case DType.DTypeEnum.INT64 =>
+                if (failOnError) {
+                  previousValue = Some(Scalar.fromLong(Math.addExact(
+                    scalar.getLong, prev.getLong)))
+                } else {
+                  previousValue = Some(Scalar.fromLong(scalar.getLong + prev.getLong))
+                }
+              case DType.DTypeEnum.FLOAT32 =>
+                previousValue = Some(Scalar.fromFloat(scalar.getFloat + prev.getFloat))
+              case DType.DTypeEnum.FLOAT64 =>
+                previousValue = Some(Scalar.fromDouble(scalar.getDouble + prev.getDouble))
+              case DType.DTypeEnum.DECIMAL32 | DType.DTypeEnum.DECIMAL64 |
+                   DType.DTypeEnum.DECIMAL128 =>
+                withResource(ColumnVector.fromScalar(scalar, 1)) { scalarCv =>
```

Comment:
nit: Could we just do what spark does on the CPU? We already know what the decimal type should be.

The following was copied from Add, and can be slightly modified to work I think.
```
private lazy val numeric = TypeUtils.getNumeric(dataType, failOnError)

checkDecimalOverflow(numeric.plus(input1, input2).asInstanceOf[Decimal], precision, scale)

  protected def checkDecimalOverflow(value: Decimal, precision: Int, scale: Int): Decimal = {
    value.toPrecision(precision, scale, Decimal.ROUND_HALF_UP, !failOnError, getContextOrNull())
  }
```

Reply:
Thanks. I have updated this.